adding electron.md for Electron beta documentation (#2446)
saucelabs/sauce-docs: docs/web-apps/automated-testing/electron.md

---
id: electron
title: Electron Support on Sauce Labs [Beta]
sidebar_label: Electron
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import useBaseUrl from '@docusaurus/useBaseUrl';

> [Electron](https://electronjs.org/) embeds Chromium and Node.js to enable web developers to create desktop applications. Sauce Labs is beta trialing support for automated testing with Electron-based apps on Windows 10 and 11, available to all users with VDC concurrency at US West or EU Central.

Reviewer: Can we change this story, it feels a bit odd. Can we make it more "customer friendly"?

## Supported Testing Platforms

Sauce Labs currently supports the following test configurations for Electron.

> ### Platforms
> Windows 10, Windows 11 (Linux and MacOS 13 support to come)
> ### Electron versions
> versions 5 - 25

Reviewer: ```suggestion
- Platforms
  - Windows 10, 
  - Windows 11 
  - (Linux and MacOS 13 support to come)
- Electron versions
  - versions 5 - 25
```
Suggestion to prevent extra headers and make it "easier" to read

## How to Get Started

### App management

> Upload a zip file containing your Electron app via [REST API](/mobile-apps/app-storage/#uploading-apps-via-rest-api). You can refer to uploaded apps by either the app_id or app_name. Note that as with mobile apps, Electron apps are accessible only to members of the same team, and retained for 60 days.

Reviewer: @lauramthomas 

The challenge we have is that we use the "Mobile Storage" for the Electron support. How should we handle this in our docs, see https://docs.saucelabs.com/mobile-apps/app-storage/?

Author: will take this as an action item to follow up on

> Upload a zip file containing your Electron app via [REST API](/mobile-apps/app-storage/#uploading-apps-via-rest-api). You can refer to uploaded apps by either the app_id or app_name. Note that as with mobile apps, Electron apps are accessible only to members of the same team, and retained for 60 days.

Reviewer: ```suggestion
Upload a zip file containing your Electron app via [REST API](/mobile-apps/app-storage/#uploading-apps-via-rest-api). You can refer to uploaded apps by either the file ID or file name. Note that as with mobile apps, Electron apps are accessible only to members of the same team, and retained for 60 days.
```

Reviewer: Suggestion would be to make the note more explicit, maybe use [admonitions](https://docs.saucelabs.com/contributing/style-guide/#admonitions)

### Binary location

> The binary location is the folderpath and filename of your Electron executable within your zip file structure.

Reviewer: ```suggestion
The `binary_location` is the folderpath and filename of your Electron executable within your zip file structure.
```

It's better to already use the variable

For example, if your zip file is structured like this:
```
SauceLabsElectronAppv1.zip

> -- [ Sauce Labs Test ]

Reviewer: Can we use common patterns for folders, so no spaces in folder names?

  -- SauceLabsElectronApp.exe
  -- mobi.dll
  -- [ resources ]
      -- ffmpeg.dll  
  [ guide ]
  -- config.yaml
  -- readme.txt
```

then the `binary` value is `'Sauce Labs Test\SauceLabsElecronApp.exe'`.

### Configuring your tests

> You need to specify Electron as the browser name along with the Electron version needed as the browser version. You will also need to include either the file_id or file_name of your uploaded zip file containing your Electron app, with the path to the binary inside that zip.

Reviewer: ```suggestion
You need to specify Electron as the [`browserName`](https://docs.saucelabs.com/dev/test-configuration-options/#browsername) along with the Electron version needed as the [`browserVersion`](https://docs.saucelabs.com/dev/test-configuration-options/#browserversion). You will also need to include either the file ID or file name of your uploaded zip file containing your Electron app, with the path to the binary inside that zip.
```

Example written in Python with an Electron app test running on Windows 11 with Chromedriver 19 at US West:

> ```

Reviewer: Can you provide the capabilities for all languages and check with the community team if they are all valid/W3C compatible?

Author: I'll work with Titus on that and get those in a future update

options = ChromeOptions()
options.set_capability('browserName', 'electron')
options.browser_version = '19'
options.platform_name = 'Windows 11'
options.binary_location='<app_folder_name>\<app_file_name.exe>'
sauce_options = {}
sauce_options['username'] = ‘<username>’
sauce_options['accessKey'] = ‘<accesskey>’

> sauce_options['app'] = 'storage:<file_id>'

Reviewer: you can use the file id or name, can we also mention that here?

Author: will add in next update

options.set_capability('sauce:options', sauce_options)
url = 'https://ondemand.us-west-1.saucelabs.com:443/wd/hub'
driver = webdriver.Remote(command_executor=url, options=options)
```

## Viewing results

> Test results are visible on the UI under “Automated Tests > Test Results.’ You can filter directly for Electron test results by selecting “Electron” from the “Browser” drop-down.

Reviewer: Can you share a screenshot to make it more clear?

## Limitations

> Electron support is currently enabled only for automated testing on Windows 10 and 11. Electron apps uploaded via REST API are not currently visible within the App Management of the UI. Live testing and UI enhancements to come in a future release.

Reviewer: ```suggestion
- Electron support is currently enabled only for automated testing on Windows 10 and 11. 
- Electron apps uploaded via REST API are not currently visible within the App Management of the UI. 
- Live testing and UI enhancements to come in a future release.
```

---
id: electron
title: Electron Support on Sauce Labs [Beta]

> sidebar_label: Electron

Reviewer 2: Just for consistency with the other frameworks:
```suggestion
sidebar_label: Using Electron
```

Else you'll end up with a redundant menu that looks like this
```
Electron
  - Electron
```

### Binary location

The `binary_location` is the folderpath and filename of your Electron executable within your zip file structure.

For example, if your zip file is structured like this:
```
SauceLabsElectronAppv1.zip
-- [ Sauce Labs Test ]
  -- SauceLabsElectronApp.exe
  -- mobi.dll
  -- [ resources ]
      -- ffmpeg.dll  
  [ guide ]
  -- config.yaml
  -- readme.txt
```

then the `binary` value is `'Sauce Labs Test\SauceLabsElecronApp.exe'`.

### Configuring your tests

You need to specify Electron as the [`browserName`](https://docs.saucelabs.com/dev/test-configuration-options/#browsername) along with the Electron version needed as the [`browserVersion`](https://docs.saucelabs.com/dev/test-configuration-options/#browserversion). You will also need to include either the file ID or file name of your uploaded zip file containing your Electron app, with the path to the binary inside that zip.
Example written in Python with an Electron app test running on Windows 11 with Chromedriver 19 at US West:

> ```

Reviewer 2: That'll use proper markups for syntax highlighting:
```suggestion
```python
```
